### PR TITLE
Implemented wave sort

### PR DESCRIPTION
**Pull request by hardik-yadav1729**
In this code I have implemented wave sort.

I hope including this code in Repository would be helpful.

Thank you for the consideration.

### CODE REVIEW

File: src/sorting/wave_sort.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/// [Wave Sort Algorithm - GeeksforGeeks](https://www.geeksforgeeks.org/sort-array-wave-form-2/)
+///
+/// # Examples
+///
+/// ```rust
+/// let array = vec![10, 90, 49, 2, 1, 5, 23];
+/// let result = wave_sort(array);
+/// // Result: [2, 1, 10, 5, 49, 23, 90]
+/// ```
+pub fn wave_sort<T: Ord>(mut arr: Vec<T>) -> Vec<T> {
```
Comment (suzp1984) [inline src/sorting/wave_sort.rs:18]: There are no reason to take the ownership of `arr`, check other sort `fn` :
```suggestion
pub fn wave_sort<T: Ord>(arr: &mut [T]) {
```

File: src/sorting/wave_sort.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    let n = arr.len();
+    arr.sort();
+
+    for i in (0..n - 1).step_by(2) {
+        arr.swap(i, i + 1);
+    }
+
+    arr
```
Comment (suzp1984) [inline src/sorting/wave_sort.rs:26]: If the sort don't take the ownership and return void. the no need to return the `arr` .

**Comment by hardik-yadav1729** [comment]
I have made the  required changes.

Please review!

**Review by suzp1984 — APPROVED** [review]
looks good!

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by siriak** [comment]
Please fix tests

**Comment by hardik-yadav1729** [comment]
I made the required changes.

Please review!

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**